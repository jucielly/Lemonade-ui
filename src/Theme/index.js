import { ThemeProvider } from "styled-components";
import PropTypes from "prop-types";

const theme = {
  colors: {
    primary: {
      light: "#B7D0B6",
      medium: "#079504",
      dark: "#026C00",
      darkest: "#013C00",
      contrast: "#ffffff",
    },
    secondary: {
      light: "#FFE58D",
      medium: "#FFDF77",
      dark: "#F7C346",
      darkest: "#DCAC0D",
      contrast: "#000000",
    },
  },
  typography: {
    font: "'Inter', sans-serif",
    sizes: {
      md: "1rem",
    },
  },
  spacing: {
    sm: "1rem",
    md: "1.25rem",
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Theme;
