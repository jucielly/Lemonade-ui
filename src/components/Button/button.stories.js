import Button from ".";

export default {
  title: "Button",
  component: Button,
  argTypes: {
    color: {
      control: {
        type: "select",
        options: ["primary", "secondary"],
      },
    },
    icon: {
      control: {
        type: "select",
        options: ["check", "refresh", "arrow-right", ""],
      },
    },
  },
};

const Template = (args) => <Button {...args} />;

export const Default = Template.bind({});

Default.args = {
  color: "primary",
  onClick: () => alert("ouch!"),
  disabled: false,
  children: "Click me!!",
};

export const Primary = Template.bind({});

Primary.args = {
  color: "primary",
  onClick: () => alert("ouch!"),
  disabled: false,
  children: "Click me!!",
};

export const Secondary = Template.bind({});

Secondary.args = {
  color: "secondary",
  onClick: () => alert("ouch!"),
  disabled: false,
  children: "Click me!!",
};

export const WithIcon = Template.bind({});

WithIcon.args = {
  color: "primary",
  icon: "check",
  onClick: () => alert("ouch!"),
  disabled: false,
  children: "Click me!!",
};

export const Disabled = Template.bind({});

Disabled.args = {
  color: "primary",
  icon: "check",
  onClick: () => alert("ouch!"),
  disabled: true,
  children: "Click me!!",
};
