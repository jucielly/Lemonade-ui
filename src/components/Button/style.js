import styled from "styled-components";

export const StyledButton = styled.button`
  font-family: ${({ theme }) => theme.typography.font};
  font-size: ${({ theme }) => theme.typography.sizes.md};
  color: ${({ theme, color }) => theme.colors[color].contrast};
  background-color: ${({ theme, color }) => theme.colors[color].dark};
  border: none;
  padding: ${({ theme }) => `${theme.spacing.md}   ${theme.spacing.sm}`};
  min-width: 311px;
  width: 100%;
  transition: 0.3s all ease-in;
  cursor: pointer;

  &:hover {
    background-color: ${({ theme, color }) => theme.colors[color].medium};
  }

  &:active {
    background-color: ${({ theme, color }) => theme.colors[color].darkest};
  }

  &:disabled {
    background-color: ${({ theme, color }) => theme.colors[color].light};
    cursor: auto;
  }

  display: flex;
  align-items: center;
  justify-content: ${({ hasIcon }) => (hasIcon ? "space-between" : "center")};
`;
