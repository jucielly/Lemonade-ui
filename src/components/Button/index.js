import { StyledButton } from "./style";
import Icon from "../Icon";
import PropTypes from "prop-types";
import React from "react";

const Button = ({ icon, children, ...props }) => {
  return (
    <StyledButton hasIcon={!!icon} {...props}>
      {children}
      {icon && <Icon icon={icon} />}
    </StyledButton>
  );
};

Button.propTypes = {
  icon: Icon.propTypes.icon,
  color: PropTypes.oneOf(["primary", "secondary"]),
  ...StyledButton.propTypes,
};

Button.defaultProps = {
  color: "primary",
};
export default Button;
