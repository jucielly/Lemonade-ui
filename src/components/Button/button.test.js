import { fireEvent, render, screen } from "@testing-library/react";
import Theme from "../../Theme";
import Button from "./index";

const renderWithTheme = (component, options) =>
  render(component, {
    wrapper: Theme,
    ...options,
  });

describe("Button", () => {
  it("Should display icon", () => {
    const icon = "arrow-right";
    renderWithTheme(<Button icon={icon} />);

    expect(screen.getByTestId(`${icon}-icon`)).toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { container } = renderWithTheme(
      <Button color="primary">Click me!</Button>
    );
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("Should be disabled", () => {
    const clickMock = jest.fn();
    renderWithTheme(
      <Button disabled onClick={clickMock}>
        Click me!
      </Button>
    );
    const buttonElement = screen.getByRole("button");
    expect(buttonElement).toHaveProperty("disabled");
    fireEvent.click(buttonElement);
    expect(clickMock).not.toBeCalled();
  });

  it("Should listen do click", () => {
    const clickMock = jest.fn();
    renderWithTheme(<Button onClick={clickMock}>Click me!</Button>);
    fireEvent.click(screen.getByRole("button"));
    expect(clickMock).toBeCalled();
  });
});
