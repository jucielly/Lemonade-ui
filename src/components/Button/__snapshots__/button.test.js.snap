// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `"<button color=\\"primary\\" class=\\"sc-bdvvtL jjBEwY\\">Click me!</button>"`;
